src/modules/claims/claims.service.ts

```diff
@@ -521,16 +521,22 @@ export class ClaimsService {
     this.validatePublishPublicClaimRequest(registrationTypes, claim);
     let url: string | undefined = undefined;
     if (registrationTypes.includes(RegistrationTypes.OnChain)) {
+      if (!claim.claimType) {
+        throw new Error(ERROR_MESSAGES.CLAIM_TYPE_MISSING);
+      }
+
       const claims = await this.getClaimsBySubject({
         did: this._signerService.did,
-        namespace: claim.claimType,
+        // get namespace from claimType
+        namespace: claim.claimType.split('.').slice(2).join('.'),
```

Comment:
This might be a bit clearer to readers as it makes it clear we want everything before `.roles.`. What do you think?

```suggestion
        namespace: claim.claimType.split('.roles.')[1],
```